### PR TITLE
initial code, example, travis

### PR DESCRIPTION
**Pull request by andyglow**
Initial code added
Tests added
Example provided
TravisCI integration enabled

### CODE REVIEW

File: src/examples/scala/ExampleRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import java.util.UUID
+
+import akka.http.documeneddsl.directives.DDirectives._
```
Comment (andyglow) [inline src/examples/scala/ExampleRoutes.scala]: oops ) fixed

File: project/Dependencies.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import sbt._
+
+object Dependencies {
```
Comment (t3hnar) [inline project/Dependencies.scala]: I can see these in two places

File: src/main/scala/akka/http/documenteddsl/directives/DDirective.scala [hunk 3]
```diff
@@ -0,0 +1,100 @@
+package akka.http.documenteddsl.directives
```
Comment (t3hnar) [inline src/main/scala/akka/http/documenteddsl/directives/DDirective.scala]: I don't like that project name does not match package

Comment (andyglow) [inline src/main/scala/akka/http/documenteddsl/directives/DDirective.scala]: renamed the project into `documenteddsl`

**Pull request merged**